fastestimator/backend/_load_model.py

```diff
@@ -49,7 +49,6 @@ def load_model(model: Union[tf.keras.Model, torch.nn.Module], weights_path: str,
     Raises:
         ValueError: If `model` is an unacceptable data type.
     """
-    assert hasattr(model, "fe_compiled") and model.fe_compiled, "model must be built by fe.build"
```

Comment:
???

Reply:
I realize our load model can be used for non-fe compiled model for convenient loading too, such as loading SAM encoder weight in the torch implementation. Removing the assertiton because of that.

Reply:
I feel like we're opening a can of worms for majority use case though. wouldn't someone who just wants to load SAM simply use model.load_state_dict() ?

Reply:
`model.load_state_dict()` won't work for multi-gpu system.   also there's an additional step of `preprocess_torch_weights` for pytorch weights too if a model is saved in mutli-gpu system but loaded in single -gpu system.  

That's why I need to call our own backend function rather than duplicating the code.

Reply:
hmm, what if you just add in fe_compiled manually to it?

Reply:
I think removing the assertion is better, because it can at least allow other pytorch models to leverage our `load_model` capability.